src/shared/condition.c

```diff
@@ -239,6 +235,44 @@ static int condition_test_kernel_version(Condition *c, char **env) {
         return true;
 }
 
+static int condition_test_version(Condition *c, char **env) {
+        const char *p = ASSERT_PTR(ASSERT_PTR(c)->parameter);
+        _cleanup_free_ char *word = NULL;
+        const char *ver, *s;
+        struct utsname u;
+        int r;
+
+        assert(c);
+        assert(c->type == CONDITION_KERNEL_VERSION || c->type == CONDITION_VERSION);
+
+        r = extract_first_word(&p, &word, " =!<>$", 0);
+        if (r < 0)
+                return log_debug_errno(r, "Failed to parse compare predicate \"%s\": %m", p);
+
+        /* An empty condition is true, but one where the right operand is missing is invalid. */
+        if (r == 0) {
+                if (!c->parameter[0])
+                        return true;
+                return log_debug_errno(SYNTHETIC_ERRNO(EINVAL), "Missing right operand in condition: %s", c->parameter);
+        }
+
+        s = strstrip(word);
+        p = c->parameter + strlen(s);
+
+        if (streq(s, "systemd")) {
+                ver = STRINGIFY(PROJECT_VERSION);
+        } else {
```

Comment:
no `{}` around single line if blocks, as per coding style

Reply:
shouldn't this rather be: 

```c
else if (STR_IN_SET(s, "kernel", "")) {
        …
}

Reply:
even if the else block has braces? doesn't look odd if the if is without braces and the else does?
```c
if (...)
    ...
else {
    ...
}
```

Reply:
If the "kernel" keyword is omitted, then "s" will contain the right operand. e.g.

```
$ build/systemd-analyze condition 'ConditionVersion= systemd > 10'
s: systemd
test.service: ConditionVersion=systemd > 10 succeeded.

$ build/systemd-analyze condition 'ConditionVersion= kernel > 10'
s: kernel
test.service: ConditionVersion=kernel > 10 failed.

$ build/systemd-analyze condition 'ConditionVersion= > 10'
s: 10
test.service: ConditionVersion=> 10 failed.
```

so how could `STR_IN_SET(s, "kernel", "")` work?